src/content/tools/pub/publishing.md

```diff
@@ -285,12 +285,14 @@ Later, when you release version `2.0.0`, it takes precedence over all
 
 As pub prefers stable releases when available, users of a prerelease package
 might need to change their dependency constraints.
-For example, if a user wants to test prereleases of version 2.1, then
+For example, if a user wants to test prereleases of version `2.1.0`, then
 instead of `^2.0.0` or `^2.1.0` they might specify `^2.1.0-dev.1`.
 
 :::note
-If a stable package in the dependency graph depends on a prerelease,
-then pub chooses that prerelease instead of a stable release.
+While `pub` prefers stable releases the same way it prefers newer versions,
+the version solver does not attempt all solutions and may pick a prerelease,
+even when a resolution that doesn't use prelreases exists.
```

Comment:
Perhaps we could expand to something like:

```suggestion
While `pub` prefers stable releases the same way it prefers newer versions,
the version solver is applying its heuristics greedily, and stops at the first solution it finds. Thus it might end up stuck at a local maximum, with a solution including prereleases even though a solution without those exists.
```

Reply:
We could do that, I wouldn't mind -- but it does get rather technical.